includes/utils/class-amp-validation-utils.php

```diff
@@ -1869,4 +1878,60 @@ public static function get_recheck_link( $post, $redirect_url, $recheck_url = nu
 		);
 	}
 
+	/**
+	 * Enqueues the block validation script.
+	 *
+	 * @return void
+	 */
+	public static function enqueue_block_validation() {
+		$slug = 'amp-block-validation';
+
+		wp_enqueue_script(
+			$slug,
+			amp_get_asset_url( "js/{$slug}.js" ),
+			array( 'jquery' ),
+			AMP__VERSION,
+			true
+		);
+
+		$data = wp_json_encode( array(
+			'i18n'     => array(
+				/* translators: %s: the name of the block */
+				'notice' => __( 'The %s block above has invalid AMP', 'amp' ),
+			),
+		) );
+		wp_add_inline_script( $slug, sprintf( 'ampBlockValidation.boot( %s );', $data ) );
+	}
+
+	/**
+	 * Adds fields to the REST API responses, in order to display validation errors.
+	 *
+	 * @return void
+	 */
+	public static function add_rest_api_fields() {
+		register_rest_field(
+			array( 'post', 'page' ),
```

Comment:
When `! amp_is_canonical()` then this should be the REST object types for all post types that have `amp` post type support. Otherwise, it should be added for all post types which have the `editor` post type support.